armory/baseline_models/pytorch/sincnet.py

```diff
@@ -54,6 +58,97 @@ def preprocessing_fn(batch):
     return np.array(processed_batch)
 
 
+def numpy_all_preprocessing_fn(batch):
+    """
+    Input is comprised of one or more clips, where each clip i
+    is given as an ndarray with shape (n_i,).
+    Preprocessing normalizes each clip and breaks each clip into an integer number
+    of non-overlapping segments of length WINDOW_LENGTH.
+    Output is a list of clips, each of shape (int(n_i/WINDOW_LENGTH), WINDOW_LENGTH)
+    """
+    if len(batch) != 1:
+        raise NotImplementedError(
+            "Requires ART variable length input capability for batch size != 1"
+        )
+    processed_batch = []
+    for clip in batch:
+        # convert and normalize
+        signal = clip.astype(np.float64)
+        signal = signal / np.max(np.abs(signal))
+
+        # break into a number of chunks of equal length
+        num_chunks = int(len(signal) / WINDOW_LENGTH)
+        signal = signal[: num_chunks * WINDOW_LENGTH]
+        signal = np.reshape(signal, (num_chunks, WINDOW_LENGTH), order="C")
+        processed_batch.append(signal)
+    # remove outer batch (of size 1)
+    processed_batch = processed_batch[0]
+    return np.array(processed_batch)
+
+
+def torch_random_preprocessing_fn(x):
+    """
+    Standardize, then normalize sound clips
+    """
+    if x.shape[0] != 1:
+        raise ValueError(f"Shape of batch x {x.shape[0]} != 1")
+    if x.dtype != torch.float32:
+        raise ValueError(f"dtype of batch x {x.dtype} != torch.float32")
+    if x.max() > 1.0:
+        raise ValueError(f"batch x max {x.max()} > 1.0")
+    if x.min() < -1.0:
+        raise ValueError(f"batch x min {x.min()} < -1.0")
+    x = x.squeeze(0)
+
+    # Signal normalization
+    x = x / x.abs().max()
+
+    # get pseudorandom chunk of fixed length (from SincNet's create_batches_rnd)
+    signal_length = len(x)
+    np.random.seed(signal_length)
+    start = int(
+        np.random.randint(signal_length / WINDOW_LENGTH - 1)
+        * WINDOW_LENGTH
+        % signal_length
+    )
+
+    x = x[start : start + WINDOW_LENGTH]
+
+    x = x.unsqueeze(0)
+    return x
+
+
+def torch_all_preprocessing_fn(x):
+    """
+    Input is comprised of one or more clips, where each clip i
+    is given as an ndarray with shape (n_i,).
+    Preprocessing normalizes each clip and breaks each clip into an integer number
+    of non-overlapping segments of length WINDOW_LENGTH.
+    Output is a list of clips, each of shape (int(n_i/WINDOW_LENGTH), WINDOW_LENGTH)
+    """
+    if x.shape[0] != 1:
+        raise NotImplementedError(
+            "Requires ART variable length input capability for batch size != 1"
+        )
+    if x.max() > 1.0:
```

Comment:
These checks are failing on adversarial inputs generated by FGSM/PGD, possibly because the Pytorchclassifier wrapper does not set clip values.